client/Contexts/ConfigFeatureFlag/ConfigFeatureFlag.spec.tsx

```diff
@@ -0,0 +1,347 @@
+/*
+ * Copyright Strimzi authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+
+import React from 'react';
+import { render, act } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+import { GET_CONFIG } from 'Queries/Config';
+import { withApolloProviderReturning, apolloMockResponse } from 'utils/test';
+import {
+  ConfigFeatureFlagProvider,
+  ConfigFeatureFlagConsumer,
+} from './Context';
+import { defaultClientConfig } from './ConfigFeatureFlag.assets';
+
+import { ConfigFeatureFlagType } from './ConfigFeatureFlag.types';
+
+describe('ConfigFeatureFlag', () => {
+  const mockClientResponse = {
+    about: {
+      version: '1.2.3',
+    },
+  };
+  const mockFeatureFlagResponse = {
+    client: {
+      Home: {
+        showVersion: true,
+      },
+      Pages: {
+        PlaceholderHome: true,
+      },
+    },
+  };
+
+  const request = {
+    query: GET_CONFIG,
+    variables: {},
+    context: {
+      purpose: 'config',
+    },
+  };
+
+  const mockError = new Error('Example error case');
+
+  const errorResponse = [
+    {
+      request,
+      error: mockError,
+    },
+  ];
+
+  const successResponse = [
+    {
+      request,
+      result: {
+        data: {
+          client: mockClientResponse,
+          featureFlags: mockFeatureFlagResponse,
+        },
+      },
+    },
+  ];
+
+  const noOpResponse = [
+    {
+      request,
+      result: {
+        data: {},
+      },
+    },
+  ];
+
+  describe('`ConfigFeatureFlagProvider` and `ConfigFeatureFlagConsumer` components', () => {
+    describe('initial state', () => {
+      it('returns the expected client configuration values', () => {
+        const expectedValues = defaultClientConfig.client;
+        const { getByText } = render(
+          withApolloProviderReturning(
```

Comment:
```suggestion
          setup(gqlMock, renderFn)
```
Where setup can create the `withApolloProvider` duplication